STYLEGUIDE.md

```diff
@@ -0,0 +1,1336 @@
+        * [Always use type parameters](#always-use-type-parameters)
+        * [Stay out of <a href="http://en.wikipedia.org/wiki/Texas-sized" rel="nofollow">Texas</a>](#stay-out-of-texas)
+        * [Avoid typecasting](#avoid-typecasting)
+        * [Use final fields](#use-final-fields)
+        * [Avoid mutable static state](#avoid-mutable-static-state)
+        * [Exceptions](#exceptions)
+           * [Catch narrow exceptions](#catch-narrow-exceptions)
+           * [Don't swallow exceptions](#dont-swallow-exceptions)
+           * [When interrupted, reset thread interrupted state](#when-interrupted-reset-thread-interrupted-state)
+           * [Throw appropriate exception types](#throw-appropriate-exception-types)
+     * [Use newer/better libraries](#use-newerbetter-libraries)
+        * [StringBuilder over StringBuffer](#stringbuilder-over-stringbuffer)
+        * [ScheduledExecutorService over Timer](#scheduledexecutorservice-over-timer)
+        * [List over Vector](#list-over-vector)
+     * [equals() and hashCode()](#equals-and-hashcode)
+     * [Premature optimization is the root of all evil.](#premature-optimization-is-the-root-of-all-evil)
+     * [TODOs](#todos)
+        * [TODOs should not reach production code](#todos-should-not-reach-production-code)
+        * [Open issues on uncompleted TODOs](#open-issues-on-uncompleted-todos)
+     * [Obey the Law of Demeter (<a href="http://en.wikipedia.org/wiki/Law_of_Demeter" rel="nofollow">LoD</a>)](#obey-the-law-of-demeter-lod)
+        * [In classes](#in-classes)
+        * [In methods](#in-methods)
+     * [Don't Repeat Yourself (<a href="http://en.wikipedia.org/wiki/Don't_repeat_yourself" rel="nofollow">DRY</a>)](#dont-repeat-yourself-dry)
+        * [Extract constants whenever it makes sense](#extract-constants-whenever-it-makes-sense)
+        * [Centralize duplicate logic in utility functions](#centralize-duplicate-logic-in-utility-functions)
+     * [Manage threads properly](#manage-threads-properly)
+     * [Avoid unnecessary code](#avoid-unnecessary-code)
+        * [Superfluous temporary variables.](#superfluous-temporary-variables)
+        * [Unneeded assignment.](#unneeded-assignment)
+     * [The 'fast' implementation](#the-fast-implementation)
+
+
+## Coding style
+
+### Formatting
+
+#### Use line breaks wisely
+There are generally two reasons to insert a line break:
+
+1. Your statement exceeds the column limit.
+
+2. You want to logically separate a thought.<br />
+Writing code is like telling a story.  Written language constructs like chapters, paragraphs,
+and punctuation (e.g. semicolons, commas, periods, hyphens) convey thought hierarchy and
+separation.  We have similar constructs in programming languages; you should use them to your
+advantage to effectively tell the story to those reading the code.
+
+#### Indent style
+We use the "one true brace style" ([1TBS](http://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS)).
+Indent size is 2 columns.
```

Comment:
while 2 spaces make you run into a situation where the last chars dont fit into the line less often i think its easier to read with 4 spaces indentation (especially since thats how iri is formatted already)

Reply:
You guys are correct. This is a "bug" in the guide and I am changing it to 4.